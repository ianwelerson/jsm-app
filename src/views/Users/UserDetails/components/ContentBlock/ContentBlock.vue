<script setup lang="ts">
export interface ContentBlockProps {
  title: string
  value: string | number
  capitalize?: boolean
}

const props = withDefaults(defineProps<ContentBlockProps>(), {
  capitalize: true,
})
</script>

<template>
  <div class="content-block">
    <p class="content-block__title">{{ props.title }}</p>
    <p
      class="content-block__value"
      :class="{ 'content-block__value--capitalize': props.capitalize }"
    >
      {{ props.value }}
    </p>
  </div>
</template>

<style lang="scss" scoped>
.content-block {
  &:not(:last-child) {
    margin-bottom: $spacing-2;

    @include screen('lg') {
      margin-right: $spacing-7;
    }
  }

  &__title {
    color: $gray-700;
    font-size: $text-xs;
    text-transform: capitalize;
  }

  &__value {
    margin: 0;
    font-size: $text-base;
    font-weight: $font-medium;

    &--capitalize {
      text-transform: capitalize;
    }
  }
}
</style>
