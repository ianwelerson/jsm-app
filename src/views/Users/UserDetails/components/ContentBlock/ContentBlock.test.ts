import { render } from '@testing-library/vue'
import ContentBlock from '@/views/Users/UserDetails/components/ContentBlock/ContentBlock.vue'

const mockProp = {
  title: 'My field',
  value: 'My value',
}

describe('ContentBlock', () => {
  it('should render content props', async () => {
    const { getByText } = render(ContentBlock, {
      props: {
        ...mockProp,
      },
    })

    getByText(mockProp.title)
    getByText(mockProp.value)
  })

  it('should match the snapshot', async () => {
    const { html } = render(ContentBlock, {
      props: {
        ...mockProp,
      },
    })

    expect(html()).toMatchSnapshot()
  })
})
