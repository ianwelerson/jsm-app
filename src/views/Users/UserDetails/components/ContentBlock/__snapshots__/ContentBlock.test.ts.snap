// Vitest Snapshot v1

exports[`ContentBlock > should match the snapshot 1`] = `
"<div class=\\"content-block\\" data-v-f09174f5=\\"\\">
  <p class=\\"content-block__title\\" data-v-f09174f5=\\"\\">My field</p>
  <p class=\\"content-block__value content-block__value--capitalize\\" data-v-f09174f5=\\"\\">My value</p>
</div>"
`;
